From a9cc20440c5acc5b8904e814ad7fc4e55bd14e52 Mon Sep 17 00:00:00 2001
From: Anilkumar Kolli <akolli@codeaurora.org>
Date: Fri, 25 Sep 2020 19:24:05 +0530
Subject: [PATCH 1/2] ath11k: fix build issues with upstream mhi
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix below build issues,

  CC [M]  drivers/net/wireless/ath/ath11k/mhi.o
drivers/net/wireless/ath/ath11k/mhi.c: In function ‘ath11k_mhi_op_status_cb’:
drivers/net/wireless/ath/ath11k/mhi.c:179:52:
error: ‘struct mhi_controller’ has no member named ‘dev’
  struct ath11k_base *ab = dev_get_drvdata(mhi_cntrl->dev);
                                                    ^
drivers/net/wireless/ath/ath11k/mhi.c:266:10:
error: ‘struct mhi_controller’ has no member named ‘link_status’

  mhi_ctrl->link_status = ath11k_mhi_op_link_status;

Signed-off-by: Anilkumar Kolli <akolli@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/mhi.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/mhi.c b/drivers/net/wireless/ath/ath11k/mhi.c
index 880650b48315..ce3911230e16 100644
--- a/drivers/net/wireless/ath/ath11k/mhi.c
+++ b/drivers/net/wireless/ath/ath11k/mhi.c
@@ -137,11 +137,6 @@ static int ath11k_pci_get_mhi_msi(struct ath11k_pci *ar_pci)
 	return 0;
 }
 
-static int ath11k_mhi_op_link_status(struct mhi_controller *mhi_ctrl)
-{
-	return 0;
-}
-
 static int ath11k_mhi_op_runtime_get(struct mhi_controller *mhi_cntrl)
 {
 	return 0;
@@ -176,7 +171,7 @@ static char *ath11k_mhi_callback_to_str(int reason)
 static void ath11k_mhi_op_status_cb(struct mhi_controller *mhi_cntrl,
 				    enum mhi_callback cb)
 {
-	struct ath11k_base *ab = dev_get_drvdata(mhi_cntrl->dev);
+	struct ath11k_base *ab = dev_get_drvdata(mhi_cntrl->cntrl_dev);
 
 	ath11k_info(ab, "mhi notify status reason %s\n",
 		    ath11k_mhi_callback_to_str(cb));
@@ -263,7 +258,6 @@ no_dt:
 	mhi_ctrl->fbc_download = true;
 	mhi_ctrl->rddm_size = ATH11K_PCI_FW_RDDM_SZ;
 
-	mhi_ctrl->link_status = ath11k_mhi_op_link_status;
 	mhi_ctrl->runtime_get = ath11k_mhi_op_runtime_get;
 	mhi_ctrl->runtime_put = ath11k_mhi_op_runtime_put;
 	mhi_ctrl->status_cb = ath11k_mhi_op_status_cb;
-- 
2.7.4

